pkg/xds/context/resources.go

```diff
@@ -146,7 +146,7 @@ func (r Resources) VirtualOutbounds() *core_mesh.VirtualOutboundResourceList {
 	return r.ListOrEmpty(core_mesh.VirtualOutboundType).(*core_mesh.VirtualOutboundResourceList)
 }
 
-func (r Resources) OtherMeshes() *core_mesh.MeshResourceList {
+func (r Resources) Meshes() *core_mesh.MeshResourceList {
```

Comment:
Why not keep the `OtherMeshes` method? That could make code less error prone.

Reply:
That'd require adding another field like `LocalMeshName string` to the Resources struct:
```go
type Resources struct {
    MeshLocalResources ResourceMap
    CrossMeshResources map[xds.MeshName]ResourceMap
}
```
Given the fact at this moment we don't have constructor for `Resources` it increases the scope of this PR.

I can maybe add method:
```go
func (r Resources) OtherMeshes(localMesh string) *core_mesh.MeshResourceList {
    ...
}
```

@lahabana @michaelbeaumont wdyt?

Reply:
I think this is great!